src/Avalonia.Controls/Notifications/NotificationCard.cs

```diff
@@ -0,0 +1,176 @@
+                });
+
+            this.GetObservable(ContentProperty)
+                .OfType<Notification>()
+                .Subscribe(x =>
+                {
+                    switch (x.Type)
+                    {
+                        case NotificationType.Error:
+                            PseudoClasses.Add(":error");
+                            break;
+
+                        case NotificationType.Information:
+                            PseudoClasses.Add(":information");
+                            break;
+
+                        case NotificationType.Success:
+                            PseudoClasses.Add(":success");
+                            break;
+
+                        case NotificationType.Warning:
+                            PseudoClasses.Add(":warning");
+                            break;
+                    }
+                });
+        }
+
+        /// <summary>
+        /// Determines if the notification is already closing.
+        /// </summary>
+        public bool IsClosing
+        {
+            get { return _isClosing; }
+            private set { SetAndRaise(IsClosingProperty, ref _isClosing, value); }
+        }
+
+        /// <summary>
+        /// Defines the <see cref="IsClosing"/> property.
+        /// </summary>
+        public static readonly DirectProperty<NotificationCard, bool> IsClosingProperty =
+            AvaloniaProperty.RegisterDirect<NotificationCard, bool>(nameof(IsClosing), o => o.IsClosing);
+
+        /// <summary>
+        /// Determines if the notification is closed.
+        /// </summary>
+        public bool IsClosed
+        {
+            get { return _isClosed; }
+            set { SetAndRaise(IsClosedProperty, ref _isClosed, value); }
+        }
+
+        /// <summary>
+        /// Defines the <see cref="IsClosed"/> property.
+        /// </summary>
+        public static readonly DirectProperty<NotificationCard, bool> IsClosedProperty =
+            AvaloniaProperty.RegisterDirect<NotificationCard, bool>(nameof(IsClosed), o => o.IsClosed, (o, v) => o.IsClosed = v);
+
+        /// <summary>
+        /// Defines the <see cref="NotificationCloseInvoked"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> NotificationCloseInvokedEvent =
+            RoutedEvent.Register<NotificationCard, RoutedEventArgs>(nameof(NotificationCloseInvoked), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="NotificationClosed"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> NotificationClosedEvent =
+            RoutedEvent.Register<NotificationCard, RoutedEventArgs>(nameof(NotificationClosed), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Raised when notification close event is invoked.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> NotificationCloseInvoked
```

Comment:
Hmm, what is the difference between `NotificationClosed` and `NotificationCloseInvoked`? What does it mean to raise an event when an event is invoked?